lib/util/template/index.ts

```diff
@@ -201,15 +213,18 @@ export function compile(
   input: CompileInput,
   filterFields = true
 ): string {
-  const data = { ...GlobalConfig.get(), ...input };
+  const childEnv = {
+    env: getChildEnv({}),
+  };
+  const data = merge({}, GlobalConfig.get(), input, childEnv);
```

Comment:
What's the proper order here? Should `input` come last so it can override the global config and the child `env`?